src/server/src/main/java/io/cassandrareaper/service/RepairScheduleService.java

```diff
@@ -77,52 +107,41 @@ public RepairSchedule storeNewRepairSchedule(
             segmentCountPerNode);
 
     scheduleBuilder.owner(owner);
+    return context.storage.addRepairSchedule(scheduleBuilder);
+  }
 
-    Collection<RepairSchedule> repairSchedules = context.storage
-        .getRepairSchedulesForClusterAndKeyspace(repairUnit.getClusterName(), repairUnit.getKeyspaceName());
+  private boolean isConflictingSchedules(Cluster cluster, RepairUnit newUnit, RepairUnit existingUnit) {
```

Comment:
Might be worth adding a comment outlining the steps to check for overlapping tables in a schedule.